processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -545,347 +547,348 @@ public void convertV8toV9(File v8Dir, File v9Dir, IndexSpec indexSpec)
+            GenericIndexed.STRING_STRATEGY
+        );
+        final GenericIndexed<String> availableMetrics = GenericIndexed.read(
+            indexBuffer, GenericIndexed.STRING_STRATEGY
+        );
+        final Interval dataInterval = new Interval(serializerUtils.readString(indexBuffer));
+        final BitmapSerdeFactory segmentBitmapSerdeFactory = mapper.readValue(
+            serializerUtils.readString(indexBuffer),
+            BitmapSerdeFactory.class
+        );
 
-      Set<String> columns = Sets.newTreeSet();
-      columns.addAll(Lists.newArrayList(dims9));
-      columns.addAll(Lists.newArrayList(availableMetrics));
-      GenericIndexed<String> cols = GenericIndexed.fromIterable(columns, GenericIndexed.STRING_STRATEGY);
-
-      final String segmentBitmapSerdeFactoryString = mapper.writeValueAsString(segmentBitmapSerdeFactory);
-
-      final long numBytes = cols.getSerializedSize() + dims9.getSerializedSize() + 16
-                            + serializerUtils.getSerializedStringByteSize(segmentBitmapSerdeFactoryString);
-      final SmooshedWriter writer = v9Smoosher.addWithSmooshedWriter("index.drd", numBytes);
-      cols.writeToChannel(writer);
-      dims9.writeToChannel(writer);
-      serializerUtils.writeLong(writer, dataInterval.getStartMillis());
-      serializerUtils.writeLong(writer, dataInterval.getEndMillis());
-      serializerUtils.writeString(writer, segmentBitmapSerdeFactoryString);
-      writer.close();
-
-      final ByteBuffer metadataBuffer = v8SmooshedFiles.mapFile("metadata.drd");
-      if (metadataBuffer != null) {
-        v9Smoosher.add("metadata.drd", metadataBuffer);
-      }
+        Set<String> columns = Sets.newTreeSet();
+        columns.addAll(Lists.newArrayList(dims9));
+        columns.addAll(Lists.newArrayList(availableMetrics));
+        GenericIndexed<String> cols = GenericIndexed.fromIterable(columns, GenericIndexed.STRING_STRATEGY);
+
+        final String segmentBitmapSerdeFactoryString = mapper.writeValueAsString(segmentBitmapSerdeFactory);
+
+        final long numBytes = cols.getSerializedSize() + dims9.getSerializedSize() + 16
+            + serializerUtils.getSerializedStringByteSize(segmentBitmapSerdeFactoryString);
+        final SmooshedWriter writer = v9Smoosher.addWithSmooshedWriter("index.drd", numBytes);
+        cols.writeToChannel(writer);
+        dims9.writeToChannel(writer);
+        serializerUtils.writeLong(writer, dataInterval.getStartMillis());
+        serializerUtils.writeLong(writer, dataInterval.getEndMillis());
+        serializerUtils.writeString(writer, segmentBitmapSerdeFactoryString);
+        writer.close();
+
+        final ByteBuffer metadataBuffer = v8SmooshedFiles.mapFile("metadata.drd");
+        if (metadataBuffer != null) {
+          v9Smoosher.add("metadata.drd", metadataBuffer);
+        }
 
-      log.info("Skipped files[%s]", skippedFiles);
+        log.info("Skipped files[%s]", skippedFiles);
 
-      v9Smoosher.close();
+      } finally {
```

Comment:
Closer docs suggest a different pattern here, see https://google.github.io/guava/releases/19.0/api/docs/com/google/common/io/Closer.html for the throwables catching.